### DISCUSSION TITLE
Extremely Fast Decision Tree (EFDT)

### DISCUSSION THREAD

**Discussion by gintriago12** [online-ml/river#715] [Q&A]
Hi,

I am getting a problem when the EFDT model has to evaluate the best split suggestion.

After evaluating 10000 samples from a dataset, in line 390 of extremely_fast_decision_tree.py, x_current is returned as 'None'. Then, in line 429 in the same file, the code evaluates x_current.merit but as x_current is None, the program stops due to an error. How can I fix it? Any suggestion is welcomed.

Best,

**Comment by smastelini** [comment]
Hi @gintriago12, thanks for reporting that. I have some ideas of what might be happening. But to be sure, I will need an MWE. Could you please provide us one?

**Reply by gintriago12** [reply]
Hi Smastelini, yeah sure. But, I apologize I don't know what exactly is an MWE, I am new to this. Is that a sort of report with the code? is it a video? I can definitely work on that.

**Reply by smastelini** [reply]
Sorry. I got used to using such acronyms, but not a long time ago I also didn't know what it meant.

It's a Minimum Working Example (MWE). It's a piece of code that will enable me to reproduce the same erroneous behavior you're observing.

**Reply by smastelini** [reply]
Without having an MWE to dive further into the code, my ideas until now include:

- A scenario where new features appear and some disappear
- A scenario where the memory management routines are wrongly interfering with the learning process

**Reply by MaxHalford** [reply]
From StackOverflow: [How to create a Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example)

**Reply by smastelini** [reply]
Hey @gintriago12, any progress with the EFDTs?
